<launch>
  <arg name="record_uwb_data" default="true" />
  <param name="wireless_clock_beacon_interval" type="int" value="100" />
  <group>
<!--
    <node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="2 0 0 0 0 0 1 map imu1004 100" />
    <node pkg="tf" type="static_transform_publisher" name="link5_broadcaster" args="0 2 0 0 0 0 1 map imu1005 100" />
    -->

    <!--  You'll need to update these to match the locations of your anchors. Anchor0 has to be in origo -->
    <node pkg="tf" type="static_transform_publisher" name="anchor0_broadcaster" args="0 0 0 0 0 0 1 map anchor0 100" />
    <node pkg="tf" type="static_transform_publisher" name="anchor1_broadcaster" args="5.03 0.00 0 0 0 0 1 map anchor1 100" />
    <node pkg="tf" type="static_transform_publisher" name="anchor2_broadcaster" args="0.00 2.81 0 0 0 0 1 map anchor2 100" />
    <node pkg="tf" type="static_transform_publisher" name="anchor3_broadcaster" args="3.90 2.81 0 0 0 0 1 map anchor3 100" />

    <remap from="/wireless_sync_stat" to="/wireless_sync_stat00"/>
    <node pkg="lps_tdoa_serial" type="tdoa_usb_anchor.py" name="tdoaserial_00" respawn="true" respawn_delay="10" output="log">
      <param name="serial_device" type="string" value="/dev/ttyUSB.tdoa_anchor00" />
    </node>

    <remap from="/wireless_sync_stat" to="/wireless_sync_stat01"/>
    <node pkg="lps_tdoa_serial" type="tdoa_usb_anchor.py" name="tdoaserial_01" respawn="true" respawn_delay="10" output="log">
      <param name="serial_device" type="string" value="/dev/ttyUSB.tdoa_anchor01" />
    </node>

    <remap from="/wireless_sync_stat" to="/wireless_sync_stat02"/>
    <node pkg="lps_tdoa_serial" type="tdoa_usb_anchor.py" name="tdoaserial_02" respawn="true" respawn_delay="10" output="log">
      <param name="serial_device" type="string" value="/dev/ttyUSB.tdoa_anchor02" />
    </node>

    <remap from="/wireless_sync_stat" to="/wireless_sync_stat03"/>
    <node pkg="lps_tdoa_serial" type="tdoa_usb_anchor.py" name="tdoaserial_03" respawn="true" respawn_delay="10" output="log">
      <param name="serial_device" type="string" value="/dev/ttyUSB.tdoa_anchor03" />
    </node>

    <node pkg="lps_tdoa_serial" type="tdoa_estimator.py" name="tdoa_estimator_00" respawn="true" respawn_delay="30" output="log">
    </node>
    <node pkg="rviz" type="rviz" name="imu_visualisation" output="screen" args="-d $(find lps_tdoa_serial)/rviz/imu4.rviz"></node>
  </group>
  
  <node pkg="rosbag" type="record" name="rosbag_record_lpsjson"
        args=" -j -o /tmp/tdoarec /synced_uwb_json /imu/data /tf"
        if="$(arg record_uwb_data)" />

</launch>
